test/test_custom_macros.jl

```diff
@@ -0,0 +1,78 @@
+module CustomMacros
```

Comment:
I think this could be a `Literate` page in the docs.

Reply:
That's a great idea!

We also should define a macro in an isolated sub-module and import it into the namespace we use for testing. This is for catching some possible namespacing issues.